### PR TITLE
feat(es/minifier): Extend inlining ident to multi usage

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

And fix an issue where swc inline global idents like
```js
var a = g
console.log(a)
```
due to side effect evalutaion order

**Related issue (if exists):**

 - Closes https://github.com/swc-project/swc/issues/5688

### CODE REVIEW

File: crates/swc/tests/tsc-references/recursiveInitializer.2.minified.js [hunk 1]
```diff
@@ -1,2 +1,2 @@
 //// [recursiveInitializer.ts]
-var n1 = n1++, n2 = n2 + n2, n3 = n3 + n3, s2 = s2 + s2, s3 = s3 + s3, b3 = !b3 || b3, b4 = !b4 && b4;
+var n1 = n1++, n2 = n2 + n2, n3 = n3 + n3, s1 = s1 + "", s2 = s2 + s2, s3 = s3 + s3, s4 = "" + s4, b1 = !b1, b2 = !!b2, b3 = !b3 || b3, b4 = !b4 && b4;
```
Comment (Austaras) [inline crates/swc/tests/tsc-references/recursiveInitializer.2.minified.js:2]: This is a regressiong, however I don't believe people would write code like this.

File: crates/swc_ecma_minifier/tests/terser/compress/inline/inline_annotation_2/output.js [hunk 2]
```diff
@@ -1 +1,2 @@
-console.log(42, NaN);
+const b = NaN;
```
Comment (Austaras) [inline crates/swc_ecma_minifier/tests/terser/compress/inline/inline_annotation_2/output.js:1]: This is also a regression, I will fix it in another PR.

**Comment by kdy1** [comment]
It failed only in concurrent mode...
Not sure what's the cause

**Comment by kdy1** [comment]
Oh, we only verify benchmarks in concurrent mode.
I think this will happen even if you use single-thread mode

**Comment by Austaras** [comment]
Luckily this can be reduced to a very simple reproduction.

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_minifier

**Pull request merged**